src/main/java/io/beanmapper/core/BeanMatch.java

```diff
@@ -41,4 +43,55 @@ public BeanMatch(Class<?> sourceClass, Class<?> targetClass, Map<String, BeanFie
     public Map<String, BeanField> getAliases() {
         return aliases;
     }
+
+    public MatchedBeanPairField findBeanPairField(String fieldName) {
+        BeanField sourceField = getSourceNode().get(fieldName);
+        if(sourceField == null) {
+            // No source field found -> check for alias
+            sourceField = getAliases().get(fieldName);
+        }
+        BeanField targetField = getTargetNode().get(fieldName);
+        if(targetField == null) {
+            // No target field found -> check for alias
+            targetField = getAliases().get(fieldName);
+        }
+        return new MatchedBeanPairField(sourceField, targetField);
+    }
+
+    public void validateMappingRequirements() {
+
+        BeanMatchValidationMessage validationMessage = null;
+        if (beanPair.isSourceStrict()) {
+            validationMessage = reportOnBeanMatchValidity(sourceNode);
+        } else if (beanPair.isTargetStrict()) {
+            validationMessage = reportOnBeanMatchValidity(targetNode);
+        }
+
+        if (validationMessage != null && validationMessage.getFields().size() > 0) {
+            throw new BeanStrictMappingRequirementsNotMetException(validationMessage);
+        }
+    }
+
+    private BeanMatchValidationMessage reportOnBeanMatchValidity(Map<String, BeanField> nodes) {
```

Comment:
Wanneer je hier eerst de validateMappingRequirements(...) aanroept, kun je hier al checken of er een mismatch is opgetreden en als dat zo is, de message maken en exception gooien! Dan maak je ook geen message wanneer er geen fout is en heb je de validationMessage en de null check hierop in bovenstaande method niet nodig.
Deze method kan dan void als return krijgen en wellicht een rename naar throwIfBeanFieldsNotMatch